File: src/main/java/eu/vok/bookstore/backend/DataService.java
package eu.vok.bookstore.backend;

import java.io.Serializable;
import java.util.Collection;

import eu.vok.bookstore.backend.data.Category;
import eu.vok.bookstore.backend.data.Product;
import eu.vok.bookstore.backend.mock.MockDataService;

/**
 * Back-end service interface for retrieving and updating product data.
 */
public abstract class DataService implements Serializable {

    public abstract Collection<Product> getAllProducts();

    public abstract Collection<Category> getAllCategories();

    public abstract void updateProduct(Product p);

    public abstract void deleteProduct(int productId);

    public abstract Product getProductById(int productId);

    public static DataService get() {
        return MockDataService.getInstance();
    }

}

File: src/main/java/eu/vok/bookstore/backend/DataService.kt
package eu.vok.bookstore.backend

import java.io.Serializable

import eu.vok.bookstore.backend.data.Category
import eu.vok.bookstore.backend.data.Product
import eu.vok.bookstore.backend.mock.VokORMDataService

/**
 * Back-end service interface for retrieving and updating product data.
 */
@Deprecated("inline")
interface DataService : Serializable {

    @Deprecated("Use EntityDataProvider")
    fun getAllProducts(): Collection<Product>

    @Deprecated("Use EntityDataProvider")
    fun getAllCategories(): Collection<Category>

    fun updateProduct(p: Product)

    fun deleteProduct(productId: Int)

    fun getProductById(productId: Int): Product?

    companion object {
        fun get(): DataService = VokORMDataService
    }
}
